File: game-prototype/src/models/actions/CommonAction.ts
export type CommonAction = {
    name: string;
    description: string;
}

type CommonActionType = 'flee';

export const CommonActions: Record<CommonActionType, CommonAction> = {
    flee: {
        name: 'Flee',
        description: 'Attempt to flee from the battle.'
    },
} as const;

File: game/models/CommonAction.cs
using Godot;

[Tool]
public partial class CommonAction : Resource {
	[Export]
	public string Name { get; set; }
	[Export]
	public string Description { get; set; }

	public CommonAction() { }

	public CommonAction(string name, string description) {
		Name = name;
		Description = description;
	}
}